internal/pkg/agent/cmd/container.go

```diff
@@ -318,6 +318,10 @@ func runContainerCmd(streams *cli.IOStreams, cfg setupConfig) error {
 		}
 		err = enroll.Wait()
 		if err != nil {
+			if errors.Is(err, &ErrDaemonReload{}) {
```

Comment:
If you are using `errors.Is` it usually means that you want to check a specific error value (which is usually a global value in some package): for example 

```
errors.Is(err, fs.ErrNotExist)
```
where [the error value is defined like this](https://pkg.go.dev/io/fs#ErrNotExist)

I see that you forced the implementation of `errors.Is` on the error type using `errors.As` below and this is a bit of a smell (you are changing the semantics of errors.Is).

The way I see we have 2 options here:
1. Define a global error value instead of a type for `ErrDaemonReload` , wrap it as the inner error along with the real error and keep using `errors.Is`
2. Keep the new type `ErrDaemonReload`, remove the `Is` custom implementation from the type and do all the checks using `errors.As` explicitly as you are asserting the type not the value